pilot/cmd/pilot-agent/main.go

```diff
@@ -307,8 +308,10 @@ var (
 			}
 
 			opts := make(map[string]interface{})
-			if sdsEnabled {
-				opts["sds_uds_path"] = sdsUdsPathVar.Get()
+			// If control plane auth is mTLS and global SDS flag is turned on, set UDS path and token path
+			// for control plane SDS.
+			if controlPlaneAuthEnabled && sdsEnabled {
+				opts["sds_uds_path"] = "unix:" + sdsUdsPathVar.Get()
```

Comment:
Commented in the other PR - either call it "sds_address" - and unix: is valid ( so is https:...), or leave it "uds_path" - then it's a path and not URL.

I prefer the first, since we'll likely support a mode where sds is remote.

Reply:
Your point is valid - I think we should make another PR to address this though. It will include the change to multiple files. It's cleaner and easier to merge later. We can add a TODO here.